Sage/carbon
# fix: fix typescript errors in declaration files

We had several typescript errors in .d.ts declaration files. These cause build errors in a project running without the skipLibCheck flag in tsconfig.

fix #5478

### Proposed behaviour

There should be no typescript errors coming from `.d.ts` files.

Install type declarations from NPM for draft-js as a dependency, the lack of which was causing an error in `text-editor.d.ts`.



### Current behaviour

If turning off `skipLibCheck: true` in `tsconfig.json` there are several errors flagged.



### Checklist




#### QA


### Additional context



### Testing instructions



To test, temporarily remove or comment out the line `"skipLibCheck": true` in `tsconfig.json`, and run `npm run type-check`. No errors should be flagged except for files inside `node_modules`.
